validate.go

```diff
@@ -80,17 +83,21 @@ var bundleValidateCommand = cli.Command{
 		}
 
 		hooksCheck := context.Bool("hooks")
-		bundleValidate(spec, rootfsPath, hooksCheck)
+		hostCheck := context.Bool("host-specific")
+		bundleValidate(spec, rootfsPath, hooksCheck, hostCheck)
 		logrus.Infof("Bundle validation succeeded.")
 		return nil
 	},
 }
 
-func bundleValidate(spec rspec.Spec, rootfs string, hooksCheck bool) {
+func bundleValidate(spec rspec.Spec, rootfs string, hooksCheck bool, hostCheck bool) {
 	checkMandatoryField(spec)
 	checkSemVer(spec.Version)
 	checkPlatform(spec.Platform)
 	checkProcess(spec.Process, rootfs)
+	if hostCheck && spec.Platform.OS == "linux" && runtime.GOOS == "linux" {
+		checkMounts(spec.Mounts)
```

Comment:
Mounts are a thing regardless of the Linux-ness of the config/host.  I'd rather have this always execute, pass the full `spec` and `hostChec` through to `supportedTypes`, and then have `supportedMountTypes` raise a skip/warning if it was asked for something it couldn't provide yet:

* `spec.Platform.OS == "windows"` → `["ntfs"], nil`
* `spec.Platform.OS == "linux" && hostCheck` → `[…stuff from /proc/filesystems…], nil`
* `spec.Platform.OS == "linux" && !hostCheck` → `nil, Skip(I don't have a generic list of Linux filesystem types)`
* `spec.Platform.OS != "linux" && spec.Platform.OS != "windows"` → `nil, Skip(I don't have a list of {OS} filesystem types)`

See opencontainers/runtime-spec#470 about `ntfs` being underspecified for Windows, but that's what the spec currently has.